### ISSUE TITLE
network effect: nobody will use this if running a full node is a prereq

### ISSUE THREAD

**Issue by sneak** [yep/eth-tweet#6]
I was interested in seeing if there is any real activity on this network - but I'm not installing a p2p app on my workstation.  Nobody will adopt this unless you have a hosted version for reading somewhere, and ideally web-based interfaces for people to create and use accounts.

**Comment by yep** [comment]
Yes, you are absolutely right. However, this is a fun project I'm running in my free time and therefore things are moving slow. That said, there is some progress. Having a [basic web based user interface](http://github.com/digitaldonkey/ethTweetUi) is a first step towards a hosted version.

Next step could be running this on a web server, maybe using something like [eth-rpc-wrapper](https://github.com/MetaMask/eth-rpc-wrapper/blob/master/server.js).

Any volunteers for preparing this setup?

**Comment by kumavis** [comment]
so this is relevant to my work at metamask

if you can make some changes in order to follow our [compatibility guide](https://github.com/MetaMask/compat-guide), I can help you get this running on our backend, even if users don't have local node running and dont have metamask installed

**Comment by d11e9** [comment]
+1 to making it zero client compatible re @kumavis suggestions.

**Comment by ptsteadman** [comment]
I'm interested in building a front-end for this.  I'm planning to look at the front ends of augur and other dapps for inspiration.  How does that compare to something like metamask?

**Comment by kumavis** [comment]
@psteadman
the ethereum dapp standard has developers build apps using the eth rpc to read from the chain and suggest txs.
[augur's app](https://app.augur.net/markets) uses their own hosted nodes by default to handle eth rpc, but from the code it looks like you can set a custom hosted node or local node. but that configuration seems to require some sort of sign in to their servers, sneaky centralization!
MetaMask's first project is a chrome extension that makes chrome run dapps like Mist (injecting web3, displaying transactions to be signed). But this only works if a user visits with Mist or the MetaMask extension. What if they don't have either? you still want your dapp to work. We've got a js-lib in development that will make your dapp work in the case where you need a fallback

if you have more questions ask in http://slack.metamask.io/

**Comment by yep** [comment]
Thank you for your feedback. 

Yes, using metamask to allow for posting tweets makes a lot of sense to me, as suggested by @kumavis.

There was the mention of a fallback js-lib being in development, does anyone know if there is any information available on this? (This should possibly be asked on metamask's slack instead). 

If @ptsteadman, @digitaldonkey or someone else is still interested in writing a front end for this or extending the solidity code, I'd be happy to discuss ideas and paper prototypes of the UI - just open a separate issue.

**Comment by digitaldonkey** [comment]
Hi, 
I learned a lot from @kumavis recently in JP. As soon as I find time I want to implement Metamask in the existing Angula2 UI. If anyone intredted in [colaboration on the existing project](https://github.com/digitaldonkey/ethTweetUi) I will provide any help I can. 

PS. If there is some UI gui wanna wireframe for it It will be cool too.

**Comment by vecvolta** [comment]
Tweet.Chain

Dear Yep, et al. (Yep and nascent team),

I read this code and I'm convinced it’s time we should transform this brilliant idea into a successful ICO: a democratic and fair tweeting company: the Tweet.Chain (or something)
Oh yeah, the Tweet.Chain. Here is a quick draft I wrote, hope you enjoy:

HAPPY NAPKIN DRAFT

Advertisers have to pay with ether for the opportunity to advertise on the Tweet.Chain platform (this is approximately 80% of revenue).
The "firehose" (other 20% of revenue), is the public data, the tweets, and is sold to companies so that they can analyze their brand, sentiment, etc.

The revenue will be distributed with these rules:

Popular users, with thousands of followers, are paid with ether. And this also applies to less popular users. The monthly payment is made proportional to the popularity on the site, based on followers. The calculation of the payment is based on a logarithmic scale, creating an elongated pyramid. This way, payment per follower will decrease, as popularity increases.
On the other hand, users that enjoy retweeting and liking other tweets, are paid for liking and retweeting.
User payment = 6 * % popularity + 4 * % tweets

Offensive content is flagged and categorized by users, which are compensated with some TweetChain tokens. Users will decide to hide content they consider offensive. The decision to block the content is made by the user. Not the Tweet.Chain staff.
Content violating copyright is being flagged by users and content owners.
Bots simulating being users are automatically removed by AI, because they affect the economy of the system.

The management and staff have a salary. They better work to improve the functionality of the service because they will be paid a percentage. The salary of the management is always a % of the income.

Democratic organization:

Tweet.Chain tokens are distributed at the ICO. Then, tokenholders can vote on specific issues that determine the future of the organization. Votes are weighted by quantity of token possession. Nevertheless, a tokenholder vote can't weigh more than 40%, even if he/she has more than such 40% tokens. This, to ensure democratic decisions.
Voting will be held each quarter
Specific issues to vote will be considered by upvoting or downvoting issues to discuss and vote.
If the tokenholders decide to do something, the management team has to do it. :o
When necessary, there will be alliances with other blockchain based companies to protect content and/or pay the content creator for tweeting their content. If tokenholders vote that way, as it will happen in important decisions.

Verification will be done with Civic or KYC.Legal for users who want their identity verified.

Tweet.Chain runs on Nem (xem), because it uses the blockchain and the organization server, therefore it can run quicker. All the images, videos, and tweets will be on the organization servers, but the payments, metrics, etc, will be on the blockchain.
At early stages, users will be allowed to link their twitter.com account to their Tweet.Chain, in a way that their tweets will be published at both platforms. This one as a temporary measure with the objective of promoting the organization growth, and facilitate adoption.

Well, I propose we start writing down the whitepaper, hiring a graphic designer for the website and then starting promoting the concept at icobench, icoreview, hacked, and finally, after feedback and wise decision making we can publish the announcement at bitcointalk.

Are you in?

Sincerely

vecvolta

APPENDIX (Yes, this napkin draft has an appendix)

CASH FLOW DIAGRAM

1° Advertisers pay ETH to Tweet.Chain
2° The monthly income is distributed paying ETH to:

Cash cushion (savings), defined by tokenholders by voting ← Every business needs one.
Tokenholder: 5% of revenue (proportional distribution) ← Because they funded the project
Management salaries 0.1% or less ← Because greed must be stopped
All staff salaries: 3% ← because it depends on the size of the platform
Infrastructure costs: It depends on each month and size of organization
Taxes (and other costs). Yep.
Users: each user is paid using a function considering their activity and popularity vs. all the active users. (Total users payment = Revenue - Cash cushion - Tokenholder cut- Management salaries - All staff salaries - infrastructure costs - taxes.) ← Because users create content, dah...
ICO MONEY WILL BE USED FOR:
1% or less = pay ico’s organization cost
38% = programming etc
1% = creation of videos to promote the platform
60% = aggressive ref link campaign “a la paypal”

Note: If necessary, there might be a “preICO”

I foresee the above numbers and structure will change dramatically, based on our first meeting. Hey! If it is not anarchist enough, then we can broaden the voting system to include all platform users (the 5% of revenue to tokenholders can survive, even with that change, because code is law, and someone has to fund the project, right?). Well, you know... it’s just a draft! LOL!
